### PR TITLE
Add git/setup action

### PR DESCRIPTION
**Pull request by reakaleek**

### CODE REVIEW

File: .github/workflows/test-google-auth.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ permissions:
   contents: read
 
 jobs:
-  test-google-auth:
+  test:
```
Comment (reakaleek) [inline .github/workflows/test-google-auth.yml:21]: Renamed this to `test` to have a single required status check named `test`

File: git/setup/action.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  steps:
+    - name: Setup git
+      shell: bash
+      run: |
+        git config --global user.name "${GIT_USER}"
+        git config --global user.email "${GIT_EMAIL}"
+        gh auth setup-git
+
+        echo "GIT_USER=${GIT_USER}" >> $GITHUB_ENV
+        echo "GIT_EMAIL=${GIT_EMAIL}" >> $GITHUB_ENV
```
Comment (reakaleek) [inline git/setup/action.yml:36]: Can this be converted to outputs or this only make sense as environment variables?

File: git/setup/action.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+      env:
+        GIT_USER: "${{ inputs.username }}"
+        GIT_EMAIL: "${{ inputs.email }}"
+        GIT_TRACE: "${{ inputs.trace }}"
```
Comment (reakaleek) [inline git/setup/action.yml:41]: Would it make sense to utilize `${{ runner.debug }}` to enable this?

File: git/setup/action.yml (hunk 2 above)
Comment (v1v) [reply git/setup/action.yml:38]: For simplicity I think env variables, then consumers don't need to care about creating them after using this opinionated action

File: git/setup/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      ...
+
+      - uses: elastic/oblt-actions/git/setup@v1
+        with:
+          username: "John"
+          email: "john@acme.com"
+          token: ${{ secrets.MY_GITHUB_PAT }}
```
Comment (v1v) [inline git/setup/README.md:30]:
```suggestion
          github-token: ${{ secrets.MY_GITHUB_PAT }}
```

File: git/setup/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# git/setup
+
+GitHub Action that setups the git username, email, and authentication with git CLI
+
+## Inputs
```
Comment (v1v) [inline git/setup/README.md:5]: Maybe a section about the env variables?

File: .github/workflows/test-git-setup.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./git/setup
+      - name: validate git user
+        run:
+          test $(git config user.name) = "obltmachine"
```
Comment (v1v) [inline .github/workflows/test-git-setup.yml:28]: and the env variables too?

Comment (reakaleek) [reply .github/workflows/test-git-setup.yml]: https://github.com/elastic/oblt-actions/pull/6/commits/4050449805aa34d46fb3fb8a098598ed22412d7b

File: git/setup/README.md (hunk 5 above)
Comment (reakaleek) [reply git/setup/README.md:5]: https://github.com/elastic/oblt-actions/pull/6/commits/e644abe1bd990f6a44a22d8e02357f3a90c6a522

**Review by v1v — APPROVED** [review]

**Review by v1v — APPROVED** [review]

**Pull request merged**